// yoastseo/src/helpers/getStemForLanguage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
	value: true
});

exports.default = function () {
	return {
		en: _determineStem.determineStem,
		de: _determineStem2.determineStem,
		nl: _determineStem3.determineStem,
		es: _stem2.default,
		fr: _stem4.default,
		ru: _stem6.default,
		it: _stem8.default
	};
};

var _determineStem = require("../morphology/english/determineStem");

var _determineStem2 = require("../morphology/german/determineStem");

var _determineStem3 = require("../morphology/dutch/determineStem");

var _stem = require("../morphology/spanish/stem");

var _stem2 = _interopRequireDefault(_stem);

var _stem3 = require("../morphology/french/stem");

var _stem4 = _interopRequireDefault(_stem3);

var _stem5 = require("../morphology/russian/stem");

var _stem6 = _interopRequireDefault(_stem5);

var _stem7 = require("../morphology/italian/stem");

var _stem8 = _interopRequireDefault(_stem7);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
